Makefile

```diff
@@ -34,22 +38,52 @@ INTERMEDIATE_CA_CRT = $(CERT_PATH)/intermediatecacrt.pem
 MFG_KEY = $(CERT_PATH)/mfgkey.pem
+COAP_CRT = $(CERT_PATH)/coapcrt.pem
+COAP_KEY = $(CERT_PATH)/coapkey.pem
 
 certificates:
 	mkdir -p $(CERT_PATH)
 	chmod 0777 $(CERT_PATH)
 	docker pull $(CERT_TOOL_IMAGE)
-	docker run --rm -v $(CERT_PATH):/out $(CERT_TOOL_IMAGE) --outCert=/out/cloudca.pem --outKey=/out/cloudcakey.pem \
-		--cert.subject.cn="ca" --cert.signatureAlgorithm=$(CERT_TOOL_SIGN_ALG) --cert.ellipticCurve=$(CERT_TOOL_ELLIPTIC_CURVE) \
-		--cmd.generateRootCA
-	docker run --rm -v $(CERT_PATH):/out $(CERT_TOOL_IMAGE) --signerCert=/out/cloudca.pem --signerKey=/out/cloudcakey.pem  \
-		--outCert=/out/intermediatecacrt.pem --outKey=/out/intermediatecakey.pem --cert.basicConstraints.maxPathLen=0 \
-		--cert.subject.cn="intermediateCA" --cert.signatureAlgorithm=$(CERT_TOOL_SIGN_ALG) \
-		--cert.ellipticCurve=$(CERT_TOOL_ELLIPTIC_CURVE) --cmd.generateIntermediateCA
-	docker run --rm -v $(CERT_PATH):/out $(CERT_TOOL_IMAGE) --signerCert=/out/intermediatecacrt.pem \
-		--signerKey=/out/intermediatecakey.pem --outCert=/out/mfgcrt.pem --outKey=/out/mfgkey.pem --cert.san.domain=localhost \
-		--cert.san.ip=127.0.0.1 --cert.subject.cn="mfg" --cert.signatureAlgorithm=$(CERT_TOOL_SIGN_ALG) \
-		--cert.ellipticCurve=$(CERT_TOOL_ELLIPTIC_CURVE) --cmd.generateCertificate
+
+	docker run \
+		--rm -v $(CERT_PATH):/out \
+		--user $(USER_ID):$(GROUP_ID) \
+		$(CERT_TOOL_IMAGE) \
+			--outCert=/out/cloudca.pem --outKey=/out/cloudcakey.pem \
+			--cert.subject.cn="ca" --cert.signatureAlgorithm=$(CERT_TOOL_SIGN_ALG) --cert.ellipticCurve=$(CERT_TOOL_ELLIPTIC_CURVE) \
+			--cmd.generateRootCA
+
+	docker run \
+		--rm -v $(CERT_PATH):/out \
+		--user $(USER_ID):$(GROUP_ID) \
+		$(CERT_TOOL_IMAGE) \
+			--signerCert=/out/cloudca.pem --signerKey=/out/cloudcakey.pem \
+			--outCert=/out/intermediatecacrt.pem --outKey=/out/intermediatecakey.pem \
+			--cert.basicConstraints.maxPathLen=0 --cert.subject.cn="intermediateCA" \
+			--cert.ellipticCurve=$(CERT_TOOL_ELLIPTIC_CURVE) --cert.signatureAlgorithm=$(CERT_TOOL_SIGN_ALG) \
+			--cmd.generateIntermediateCA
+
+	docker run \
+		--rm -v $(CERT_PATH):/out \
+		--user $(USER_ID):$(GROUP_ID) \
+		$(CERT_TOOL_IMAGE) \
+			--signerCert=/out/intermediatecacrt.pem --signerKey=/out/intermediatecakey.pem \
+			--outCert=/out/mfgcrt.pem --outKey=/out/mfgkey.pem --cert.san.domain=localhost \
+			--cert.san.ip=127.0.0.1 --cert.subject.cn="mfg" \
+			--cert.signatureAlgorithm=$(CERT_TOOL_SIGN_ALG) --cert.ellipticCurve=$(CERT_TOOL_ELLIPTIC_CURVE) \
+			--cmd.generateCertificate
+
+	docker run \
+		--rm -v $(CERT_PATH):/out \
+		--user $(USER_ID):$(GROUP_ID) \
+		${CERT_TOOL_IMAGE} \
+			--signerCert=/out/cloudca.pem --signerKey=/out/cloudcakey.pem \
+			--outCert=/out/coapcrt.pem --outKey=/out/coapkey.pem \
+			--cert.san.ip=127.0.0.1 --cert.san.domain=localhost \
+			--cert.signatureAlgorithm=$(CERT_TOOL_SIGN_ALG) --cert.ellipticCurve=$(CERT_TOOL_ELLIPTIC_CURVE) \
+			--cmd.generateIdentityCertificate=$(CLOUD_SID)
```

Comment:
to test cloud certificate use:
`--cmd.generateCertificate --cert.subject.cn=uuid:$(CLOUD_SID)` instead of `--cmd.generateIdentityCertificate`